sdv/constraints/tabular.py

```diff
@@ -941,20 +908,50 @@ def _transform(self, table_data):
                 Transformed data.
         """
         low = table_data[self.low_column_name]
+        middle = table_data[self.middle_column_name]
         high = table_data[self.high_column_name]
 
-        data = logit(table_data[self.middle_column_name], low, high)
-        table_data[self._transformed_column] = data
-        table_data = table_data.drop(self.middle_column_name, axis=1)
+        if self._is_datetime:
+            diff_column_1 = get_datetime_diff(middle, low, self._dtype)
+            diff_column_2 = get_datetime_diff(high, middle, self._dtype)
+        else:
+            diff_column_1 = middle - low
+            diff_column_2 = high - middle
 
-        return table_data
+        self.low_diff_column_name = modify_existing_name(
+            self.low_diff_column_name, table_data.columns
+        )
+        self.high_diff_column_name = modify_existing_name(
+            self.high_diff_column_name, table_data.columns
+        )
+
+        table_data[self.low_diff_column_name] = np.log(diff_column_1 + 1)
+        table_data[self.high_diff_column_name] = np.log(diff_column_2 + 1)
+
+        list_columns_nans = [self.low_column_name, self.middle_column_name, self.high_column_name]
+        nan_column = compute_nans_column(table_data, list_columns_nans)
+        if nan_column is not None:
+            self.nan_column_name = modify_existing_name(nan_column.name, table_data.columns)
+            table_data[self.nan_column_name] = nan_column
+            if self._is_datetime:
+                mean_value_low = table_data[self.low_column_name].mode()[0]
+            else:
+                mean_value_low = table_data[self.low_column_name].mean()
+
+            table_data = table_data.fillna({
+                self.low_column_name: mean_value_low,
+                self.low_diff_column_name: table_data[self.low_diff_column_name].mean(),
+                self.high_diff_column_name: table_data[self.high_diff_column_name].mean()
+            })
```

Comment:
This chunk is pretty similar to what's in the `Inequality` constraint. Maybe there's a way for us to reuse the logic

Reply:
I agree, I tried something I didn't push because I wanted to discuss it with you quickly. The limitation here for me is the datetime for the first column. There are workarounds.